src/Compiler/OnnxMlirCompiler.cpp

```diff
@@ -7,14 +7,47 @@
 
 extern "C" {
 namespace onnx_mlir {
-ONNX_MLIR_EXPORT int omCompileFromFile(const char *inputFilename,
+
+ONNX_MLIR_EXPORT int64_t omSetCompilerOptionsFromEnv(const char *envVarName) {
+  // ParseCommandLineOptions needs at least one argument
+  std::string nameStr = "program-name";
+  const char *argv[1];
+  argv[0] = nameStr.c_str();
+  const char *name = envVarName ? envVarName : OnnxMlirEnvOptionName.c_str();
+  return llvm::cl::ParseCommandLineOptions(
+      1, argv, "SetCompilerOptionsFromEnv\n", nullptr, name);
+}
+
+ONNX_MLIR_EXPORT int64_t omSetCompilerOptionsFromArgs(
+    int64_t argc, char *argv[]) {
+  return llvm::cl::ParseCommandLineOptions(
+      argc, argv, "SetCompilerOptionsFromEnv\n");
+}
+
+ONNX_MLIR_EXPORT int64_t omSetCompilerOptionsFromArgsAndEnv(
+    int64_t argc, char *argv[], const char *envVarName) {
+  const char *name = envVarName ? envVarName : OnnxMlirEnvOptionName.c_str();
+  return llvm::cl::ParseCommandLineOptions(
+      argc, argv, "SetCompilerOptionsFromEnv\n", nullptr, name);
+}
+
+ONNX_MLIR_EXPORT int64_t omSetCompilerOptions(
+    const OptionKind kind, const char *val) {
+  return setCompilerOption(kind, std::string(val));
+}
+
+ONNX_MLIR_EXPORT const char *omGetCompilerOption(const OptionKind kind) {
+  std::string val = getCompilerOption(kind);
+  return val.c_str();
+}
+
+ONNX_MLIR_EXPORT int64_t omCompileFromFile(const char *inputFilename,
```

Comment:
@sstamenova  I can add it back, but it will be misleading in that Options are set "forever", so they are not limited to this call. Let me know if that make you change your mind. Of course, we could save the old values, set the new one, and then reset them at the end. I wonder if it is worth the effort. 
Let me know if this changed your mind

Reply:
That's fine as long as @MikeHolman is ok with it as well.

Reply:
Seems reasonable enough to me.